src/inmanta/plugins.py

```diff
@@ -52,6 +52,7 @@
 from inmanta.warnings import InmantaWarning
 
 T = TypeVar("T")
+T_FUNC = TypeVar("T_FUNC", bound=Callable[..., Any])
```

Comment:
```suggestion
T_FUNC = TypeVar("T_FUNC", bound=Callable[..., object])
```

I *think* this should work equally well, and it's more correct. I'm in favor of doing the same for the cache decorator, but less opinionated about that because it was already predefined there.

Reply:
If this doesn't work out, feel free to ignore.

Reply:
It looks to me like it works!